Squeezer/src/main/java/uk/org/ngo/squeezer/service/RandomPlayDelegate.java

```diff
@@ -7,20 +7,19 @@
 
 import uk.org.ngo.squeezer.model.Player;
 
-// Has to be instantiated once to have the mDelegate
 public class RandomPlayDelegate {
 
     private static final String TAG = "RandomPlayDelegate";
 
-    public RandomPlayDelegate(SlimDelegate mDelegate) {
-        if (RandomPlayDelegate.delegate == null) {
-            RandomPlayDelegate.delegate = mDelegate;
+    RandomPlayDelegate(SlimDelegate mDelegate) {
+        if (RandomPlayDelegate.slimDelegate == null) {
+            RandomPlayDelegate.slimDelegate = mDelegate;
         }
     }
 
-    static public SlimDelegate delegate; // Make it final if possible
+    private static SlimDelegate slimDelegate;
```

Comment:
This is still static, this is bad practice. Either make slimDelegate and the methods not-static or add slimDelegate as a parameter to the methods.

Reply:
Using non-static now.